### ISSUE TITLE
Issues with BuildOptions

### ISSUE THREAD

**Issue by Walobyte** [microsoft/vscode-azure-iot-edge#529]
While attempting to recreate what was deemed 'Closed' here: microsoft/vscode-azure-iot-edge#95

My goal is to have an azure yaml pipeline pass an (argument) access token into my docker file so that it can complete a dotnet restore with the provided arguments (essentially to authenticate to a private nuget feed).

(for testing purposes, I have this working with a system access token hard coded as an ARG value in the docker file.)

With standalone docker, this should be faily simply, just define the docker command

docker build --build-arg some_variable_name=a_value

From module.json buildOptions:

"buildOptions":[ "--build-args some_variable_name=${a_value_placeholder}" ]

${a_value_placeholder} is described in my .env file which would be populated in my yaml pipeline.

When iotedge calls docker build, no arguments are supplied.

docker build --rm "C\...\dockerfile" -t localhost:5000/module:0.0.1

Any insights?

**Comment by czgtest** [comment]
@Walobyte Thank you for bringing this to our attention, We will investigate it and discuss when to fix it.
![image](https://user-images.githubusercontent.com/35018765/74921157-51f98500-5408-11ea-87e1-20183d762285.png)

**Comment by czgtest** [comment]
@Walobyte  Any update from you,  do you try this solution and resolve this issue ?

**Comment by Walobyte** [comment]
@czgtest  My apologies on the delayed response. 

Tl;dr: Yes, your provided solution worked.

Full explanation: 

I am building on arm32v7 platform. I took the assumption that removing the other platforms from the module.json and removing the unneeded dockerfiles would be sufficient as shown here:

![image](https://user-images.githubusercontent.com/60037622/75677108-b1e9fa00-5c58-11ea-9363-853307e9b9da.png)


However, I was still unable to get it to build correctly. I later found at the bottom of vs code, there is a drop down to select the default platform, even though my solution only contains 1 available platform as shown here:

![image](https://user-images.githubusercontent.com/60037622/75676523-67b44900-5c57-11ea-88cb-63bd075c741b.png)

Once I modified the default target platform, I was successful in building the solution as shown here: 

![image](https://user-images.githubusercontent.com/60037622/75677019-7cdda780-5c58-11ea-8b3e-dc2733070fc9.png)

Lastly, I am still unable to get this to build in VS 2019 following the steps I provided in this post. For now, I have switched to only using vs code going further. I understand this github issue is targeted to vs code, so I will open a ticket else where. 

Additionally, is there a MS documentation on this subject? The only available sources I could find are through github issues/pull requests. It would be nice to have this further documented. This was a rather difficult issue to overcome such a simple task, hopefully this post can provide others with additional help.

Thank you for your assistance.

**Comment by czgtest** [comment]
@Walobyte  here is some doc for using edge extension in VS2019 .
[Use Visual Studio 2019 to develop and debug modules for Azure IoT Edge](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-visual-studio-develop-module)
[Easily develop and debug Azure IoT Edge C modules with Azure IoT Edge Tools](https://devblogs.microsoft.com/iotdev/easily-develop-and-debug-azure-iot-edge-c-modules-with-azure-iot-edge-tools-preview-0-3-1/)
if you want to build on arm32v7 platform , you can create arm32v7 project in VS2019
![image](https://user-images.githubusercontent.com/35018765/75739941-49c31300-5d41-11ea-96d2-4a4f0bd74393.png)
you also can change the platform in your project property 
then right-click the project and select build and push IoT edge Modules
![image](https://user-images.githubusercontent.com/35018765/75740048-99094380-5d41-11ea-854d-a5a3329e3b0e.png)

**Comment by czgtest** [comment]
@Walobyte,  if BuildOptions issue is resolved ,  you can close this issue,
if you have more questions about using VS2019,  you can open new issue here https://github.com/microsoft/vs-azure-iot-edge-docs.
